tests/PhpUnit/GuzzleAssertsTraitTest.php

```diff
@@ -206,6 +207,31 @@ public function testAssertRequestHeaderDoesNotMatch()
         }
     }
 
+    public function testAssertRequestQueryDoesNotMatch()
+    {
+        $query = [
+            'tags' => ['foo', '1'],
+        ];
+
+        $request = $this->createMockRequest('GET', '/api/pets', $this->getValidHeaders(), $this->getValidRequestBody(), $query);
+
+        try {
+            self::assertRequestMatch($request, $this->schemaManager);
+            self::fail('Expected ExpectationFailedException to be thrown');
+        } catch (ExpectationFailedException $e) {
+            self::assertEquals(
+                <<<EOF
+Failed asserting that {"tags":["foo",1]} is a valid request query.
+[limit] The property limit is required
+[tags[1]] Integer value found, but a string is required
```

Comment:
That's the issue with query string validation: the query string as per the RFC is not typed. But we want to do some validation based on the type (int max/min per example) and in the userland code the query string parameters are typed. For now I'm using a "magic method" to transtype the parameters as best as possible but it's not perfect and we may have some cases misshandled.
The question now is whether it is acceptable or we must try to find another way (maybe using a combination of this type detection and another method using the type defined in the json schema? But it may be complicated)